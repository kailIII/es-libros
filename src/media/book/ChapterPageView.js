import React, { Component } from 'react';
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'

import markdownToComponentArray from '../../markdown'
import ProgressWheel from '../../comp/ProgressWheel'

const NextChapterButton = (props) => {
  const nextChapterUrl  = props.nextChapterUrl
  if (nextChapterUrl)
    return (
      <Link className="next-chapter" to={nextChapterUrl}>
        Siguiente Capitulo
      </Link>
    )
  return null
}

const ChapterPageContent = (props) => {
  const {
      text,
      title,
      nextChapterUrl,
    } = props

    return (
      <div>
        <div className="container-markdown">
          <h2 className="book-markdown">{title}</h2>
          { markdownToComponentArray(text, 'book-markdown') }
        </div>
        <NextChapterButton nextChapterUrl={nextChapterUrl} />
      </div>
    )
}

export default class ChapterPageView extends Component {

  render() {
    const {
      pageData
    } = this.props

    if (pageData)
      return <ChapterPageContent {...pageData} />
    else
      return <ProgressWheel />
  }
}

ChapterPageView.propTypes = {
  pageData: PropTypes.shape({
    nextChapterUrl: PropTypes.string,
    text: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
  })
}
