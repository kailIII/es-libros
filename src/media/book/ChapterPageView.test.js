import React from 'react';
import renderer from 'react-test-renderer';
import ChapterPageView from './ChapterPageView.js';
import {
  BrowserRouter as Router,
} from 'react-router-dom'

const assertViewMatchesSnapshot = (view) => {
  const component = renderer.create(view)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
}

it('renders a normal chapter text without crashing', () => {
  const props = {
    pageData: {
      nextChapterUrl: '/book/0/2',
      text: 'This is a chapter.',
      title: 'Chapter 1',
    }
  }
  const view = <Router><ChapterPageView { ...props} /></Router>
  assertViewMatchesSnapshot(view)
});

it('renders a final chapter text without crashing', () => {
  const props = {
    pageData: {
      text: 'This is a chapter.',
      title: 'Chapter 1',
    }
  }
  const view = <ChapterPageView { ...props} />
  assertViewMatchesSnapshot(view)
});

it('renders a progress wheel if there is no page data', () => {
  const props = { }
  const view = <ChapterPageView { ...props} />
  assertViewMatchesSnapshot(view)
});
