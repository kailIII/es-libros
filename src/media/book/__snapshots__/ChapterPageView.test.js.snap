exports[`test renders a final chapter text without crashing 1`] = `
<div>
  <div
    className="container-markdown">
    <h2
      className="book-markdown">
      Chapter 1
    </h2>
    <p
      className="book-markdown">
      This is a chapter.
    </p>
  </div>
</div>
`;

exports[`test renders a normal chapter text without crashing 1`] = `
<div>
  <div
    className="container-markdown">
    <h2
      className="book-markdown">
      Chapter 1
    </h2>
    <p
      className="book-markdown">
      This is a chapter.
    </p>
  </div>
  <a
    className="next-chapter"
    href="/book/0/2"
    onClick={[Function]}>
    Siguiente Capitulo
  </a>
</div>
`;

exports[`test renders a progress wheel if there is no page data 1`] = `
<div
  className="loader"
  style={Object {}} />
`;
